src/firefly/js/visualize/ui/FileUploadViewPanel.jsx

```diff
@@ -72,46 +57,57 @@ const TABLES_ONLY_SUPPORTED_TYPES=[
     FileAnalysisType.Table,
 ];
 
-
-const isTablesOnly= () => getAppOptions()?.uploadPanelLimit==='tablesOnly';
-
-
 const uploadOptions = 'uploadOptions';
 
-let currentAnalysisResult, currentReport, currentSummaryModel, currentDetailsModel;
 const FILE_UPLOAD_KEY= 'file-upload-key-';
 let keyCnt=0;
 
-export function FileUploadViewPanel({setSubmitText}) {
+export function FileUploadViewPanel({setSubmitText, acceptMoc}) {
+
+    const {groupKey}= useContext(FieldGroupCtx);
 
-    const {isLoading,statusKey} = useStoreConnector(() => getComponentState(panelKey, {isLoading:false,statusKey:''}));
+    const {isLoading,statusKey} = useStoreConnector(() => getComponentState(groupKey, {isLoading:false,statusKey:''}));
     const isWsUpdating          = useStoreConnector(() => isAccessWorkspace());
-    const uploadSrc             = useStoreConnector(() => getFieldVal(panelKey, uploadOptions));
-    const {message, analysisResult, report, summaryModel, detailsModel} = useStoreConnector((oldState) => getNextState(oldState));
+    const uploadSrc             = useStoreConnector(() => getFieldVal(groupKey, uploadOptions));
+
+
+    const SUMMARY_TBL_ID = groupKey; //FileUploadAnalysis
+    const DETAILS_TBL_ID = groupKey + '-Details';
+    const UNKNOWN_FORMAT = 'UNKNOWN';
+    const summaryUiId = SUMMARY_TBL_ID + '-UI';
+    const detailsUiId = DETAILS_TBL_ID + '-UI';
+
+    let {message, analysisResult, report, summaryModel, detailsModel, prevAnalysisResult} = useStoreConnector((oldState) =>
+        getNextState(oldState, groupKey));
```

Comment:
If you do what I suggested in the comment below I don't think you will need `useStoreConnector` any more

It should all me more straight forward.


```js
const [getUploadMetaInfo, setUploadMetaInfo]= useFieldGroupMetaState({summaryModel:undefined, analysisResult: undefined, etc etc })
let {message, analysisResult, report, summaryModel, detailsModel, prevAnalysisResult} = getNextState(setUploadMetaInfo() oldState));
```